### ISSUE TITLE
Transition to theme removes useful TypeScript suggestion/autocompletions from component props

### ISSUE THREAD

**Issue by policiu** [themesberg/flowbite-react#759]
**Describe the bug**
Props such as "size" in components such as "Button" do allow for typescript suggestions/autocomplete.

**To Reproduce**

1. In an IDE, create a variable which goes to a flowbite-react component (e.g. `<Button />`
2. Add a prop such as `size`
3. Try to see what the options are without going to the type.

**Expected behavior**
Typings work. Without them, we lose autocomplete and any compiler checks.

**Screenshots**
Bug:
<img width="426" alt="image" src="https://github.com/themesberg/flowbite-react/assets/1704165/03a7ed88-1ad9-43e9-950e-759c25c1ad93">

Expected:
<img width="340" alt="image" src="https://github.com/themesberg/flowbite-react/assets/1704165/148ebeed-56f0-4fff-84e7-0621bc4ddbfc">

**System information:**

Probably not indented, but here is my VSCode info.

- Device: PC 
- Resolution: N/A 
- OS: Windows 11
- Browser: Visual Studio Code (?) Chrome
- Version: 1.78.2

**Project information:**

- Tailwind: 3.3.2
- Flowbite: 1.6.5
- Flowbite React: 0.4.4
- Type: next.js (?)

**Additional context**

This is complicated since I'm not sure the vision of themes, but ideally, I would want the defaults to work at least.

My guess on the vision is to be able to define _new_ types and then be able to use them in component. Typings for this is fairly complicated and I couldn't find an easy way in TypeScript to help.
Here is an example of autocompleting the default types and allowing custom. Note that the custom types do not get suggested.  
 
[Playground link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwBQdA9I3ACrIDWWAdnESXABiAGwgB3AEbAYWVHSwAPSLDjBuMqATxYhoydKwBlYAC9ZcAN504NuAtQAuOKhhQ1Ac3q3nIJy7fcnta2IAAmfq4eXrbC7hEBQd4KwvFRwTYA5ABMyRmpgdGZAMy5+Ym2GQAspc6RBelwGQCsNf5p3hkAbK115ZkA7D0J9AC+DIrK8GoaWrg6AEIArjAwENzGZuiKMtyh6AAKwLgcADwi4lIyG7IANI32GXAAPo2oII8vGbEf98IZAHw2KzeADaXAAnmUALplUbjJTQeAwcFgHQAESwWDA+2QsGAyGEJ1YgIAvGw4NseHs4BAJAArLD4OAAfksDRB+1UvAhEAIbChzKcGKxOLxBKJHKh-1GcCcrHozApuzgME4PAYiuwIAgADdzDAABY6MhgrCQlDccEwi3gsh8YikZAqlFYOjI1GYLDavUASV2iiJpLZoIA0ly4Dy+awUOg2oEKQodtSw6zuFg9VBZXBuIsQBIsJnKbt0Cns+mC1mQ9bWCCq6N6EwWABZNAcZ2ot0uuBLFZra6oDFaRbCeBkrW6rB+0IBnurdamWTShh0aYF2YLZZz-bEMDoOjA2yMBBIQyLdzuWTwCCLTPTocj1B3CTLFDCPQ2w0eCnCVA6BCMBpUAXJxI27Tc+wXAcsHveAXgsMYxkbIQ1TYI0QFdVdNG0MDe24Vg0J0A8bCAswnFnCDNjoRDcDWFw4FwRYXBIfCvSwIVMWxXEYHxQlyLwgjATgMkiOcYDg28GI4nIWIyBuBpvCPJAAEFQlCL8nTINMxDtEisAAOjgH0+S00TNhQShuAyeACT0LBQjuQ0eDgMQdFQA1r2EUI4EoAgK1WFUjTgCQ0B0c59CuSDEAAiTbGQMAwGEWQnDIXBcWIGBUFkhoEPhSZ6No+A+KcbA8BgPTBAAYROPjtwgXcgwACjAHdHBwrcWoASiE-59waGjuDoixdJGIS4GaurUAbbxKBgG9eBOQETkYJcqI1Y84CbWR0DEaQDQC1yFzgI1KCiuh+roxRkHARLRuq8DeF0kkyAACTNOBRAgDgAEI4AAOQgCkoGIKBUC+u1lroIA)

**Comment by tulup-conner** [comment]
Yeah, the problem is that we have created typings for each of these props - for example, `size` here - that allow you to add any arbitrary size you like. Unfortunately, TypeScript doesn't let you do that and also simultaneously provide autofill for the default values. 

There might be a solution to that I don't know about. I would absolutely love to fix this, because I also find it unintuitive.

**Comment by policiu** [comment]
Yeah, to clarify, we can make the default types autofill and allow custom types, but the trade off is that the custom types do not auto fill (see playground link above) and it adds some typing silliness. Although, this is also a bit unintuitive. (E.g., "xl" would auto fill, but "myCustomValue" would not in "size"). 

I don't think there a good solution for auto filling both :/ (you _could_ if you require filling out the entirety of a Custom Theme with Templates, but I really don't think that's a good idea)
